File: app/src/main/java/com/vikingsen/cheesedemo/job/AppJobCreator.java
package com.vikingsen.cheesedemo.job;

import com.evernote.android.job.Job;
import com.evernote.android.job.JobCreator;

import javax.inject.Inject;
import javax.inject.Provider;
import javax.inject.Singleton;

import timber.log.Timber;

@Singleton
public class AppJobCreator implements JobCreator {

    private final Provider<CommentSyncJob> commentSyncJobProvider;

    @Inject
    AppJobCreator(Provider<CommentSyncJob> commentSyncJobProvider) {
        this.commentSyncJobProvider = commentSyncJobProvider;
    }

    @Override
    public Job create(String tag) {
        switch (tag) {
            case CommentSyncJob.TAG:
                return commentSyncJobProvider.get();
            default:
                Timber.w("Cannot find job for tag [%s]. Be sure to add Job to AppJobCreator", tag);
                return null;
        }
    }
}

File: app/src/main/java/com/vikingsen/cheesedemo/job/AppJobCreator.kt
package com.vikingsen.cheesedemo.job

import com.evernote.android.job.Job
import com.evernote.android.job.JobCreator
import timber.log.Timber
import javax.inject.Inject
import javax.inject.Provider
import javax.inject.Singleton

@Singleton
class AppJobCreator
@Inject constructor(private val commentSyncJobProvider: Provider<CommentSyncJob>) : JobCreator {

    override fun create(tag: String): Job? = when (tag) {
        CommentSyncJob.TAG -> commentSyncJobProvider.get()
        else -> {
            Timber.w("Cannot find job for tag [%s]. Be sure to add Job to AppJobCreator", tag)
            null
        }
    }
}
